mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1559,13 +1575,17 @@ private void executeBatchUpdate(Map<MongoCollection<Document>, BatchInsertionTas
 				if ( SnapshotType.INSERT == tuple.getSnapshotType() ) {
 					Document document = getCurrentDocument( snapshot, insertStatement, entityKey );
 					insertStatement = objectForInsert( tuple, document );
+					//process gridfs
+					GridFsUtil.storeContentToGridFs( currentDB,insertStatement,entityKey,optionService,tuple, tuple.getSnapshotType() );
```

Comment:
This doesn't look like the right place to store the content to GridFS. Shouldn't this happen when we flush the inserts?
See method `flushInserts`.

I would expect something similar for the updates

Reply:
ok. I will see

Reply:
Do you mean method
```
private static void flushInserts(Map<MongoCollection<Document>, BatchInsertionTask> inserts)
```
?

If yes .... Then you make mistake.

Reply:
I think you right. I will try to move GridFS feature to the method.

Reply:
To clarify, what I meant is that we only execute the insert when we called the `flushInserts`  method and everything before that is in preparation for the insert. While you are already saving something in GridFS before is flushed.